src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation valueOfCode(String useCode) {
+            for (Operation o: values()) {
+                if (o.getCode().equals(useCode)) {
+                    return o;
+                }
+            }
+            return null;
```

Comment:
Returning null in valueOfCode method might lead to potential NullPointerException. Consider throwing a exception with a proper error message instead of returning null.